docs: split some policies by the implicated method (#14211)
conan-io/conan-center-index: docs/adding_packages/sources_and_patches.md

# Sources and Patches

> This documents contains everything related to the `source()`. This includes picking sources, where they should come from and goes into when and how to modify sources.

Reviewer: ```suggestion
This documents contains everything related to the `source()`. This includes picking sources, where they should come from.
```
in another section, we already stated sources are immutable

> This is a very important aspects and it helps us to establish the quality of the packages offered by ConanCenter.

Reviewer: ```suggestion
These are a very important aspects and it helps us to establish the quality of the packages offered by ConanCenter.
```
aspects - plural

<!-- toc -->
## Contents

  * [Sources](#sources)<!-- endToc -->

## Sources

**Origin of sources:**

* Library sources should come from an official origin like the library source code repository or the official
release/download webpage.

> * If an official source archive is available, it should be preferred over an auto-generated archive.

Reviewer: maybe we should explain it a little bit more, on how to distinguish these two.
e.g., in GitHub projects, prefer archives from the `releases` tab, rather `code` - `download zip`

> **Source immutability:** Downloaded source code stored under `source` folder should not be modified. Any patch should be applied to the copy of this source code when a build is executed (basically in `build()` method).

Reviewer: ```suggestion
**Source immutability:** Downloaded source code stored under `source` folder should not be modified. Any patch or `replace_in_file` statement should be applied to the copy of this source code when a build is executed (basically in `build()` method).
```

> **Building from sources:** Recipes should always build packages from library sources.

Reviewer: we have some known exceptions to this rule (MSYS2, Android NDK, etc.)

**Sources not accessible:**

* Library sources that are not publicly available will not be allowed in this repository even if the license allows their redistribution.

> * If library sources cannot be downloaded from their official origin or cannot be consumed directly due to their

Reviewer: we had some special cases for that, e.g. for `gSOAP`. maybe explain it here.
